packages/components/src/guide/index.js

```diff
@@ -57,6 +59,20 @@ export default function Guide( {
 		}
 	};
 
+	useEffect( () => {
+		// Keeping the focus within the guide when the page changes
+		// prevents the modal from closing and avoids focus loss.
```

Comment:
It seems in React 17, the "Guide" component was relying a buggy behavior in `useFocusOutside` that prevented it from closing the modal even if the focus moved outside (focus loss). 

The change here prevents the focus loss in the guide component when clicking "previous" on the second page of the guide which solves the e2e test failure I had here.

cc @talldan @noisysocks who might know better if there's anything I'm missing here.

Reply:
That makes sense. I remember a discussion about avoiding focus loss on the 'next' button when it changes into the finish button, but not so much about the 'previous' button. Maybe that was missed in the original implementation.